core/shared/src/main/scala/fs2/Stream.scala

```diff
@@ -4374,15 +4376,15 @@ object Stream extends StreamLowPriority {
           s: Stream[Pure, O],
           init: () => B
       )(foldChunk: (B, Chunk[O]) => B, finalize: B => C): C =
-        finalize(Compiler.compile(s.covary[IO].free, init())(foldChunk).unsafeRunSync)
+        finalize(Compiler.compile(s.covary[SyncIO].free, init())(foldChunk).unsafeRunSync)
```

Comment:
These were bugs, which caused exceptions on Scala.js. I think we had some tests that weren't running before, only looked like they were running.